src/server/game/Entities/GameObject/GameObject.cpp

```diff
@@ -706,9 +706,18 @@ void GameObject::Update(uint32 diff)
                 //! The GetOwnerGUID() check is mostly for compatibility with hacky scripts - 99% of the time summoning should be done trough spells.
                 if (GetSpellId() || GetOwnerGUID())
                 {
-                    SetRespawnTime(0);
-                    Delete();
-                    return;
+                        //Don't delete spell spawned chests, which are not consumed on loot
+                        if (m_respawnTime > 0 && GetGoType() == GAMEOBJECT_TYPE_CHEST && !GetGOInfo()->IsDespawnAtAction())
+                        {
+                                UpdateObjectVisibility();
+                                SetLootState(GO_READY);
+                        }
+                        else
+                        {
+                                SetRespawnTime(0);
+                                Delete();
+                        }
+                        return;
```

Comment:
isn't a bit too much space for indentation? we usually use 4 spaces

Reply:
Fair enough, I just lined up as with tabs... I changed it to 4 spaces now :smile: